core/tee/tee_svc_cryp.c

```diff
@@ -3402,54 +3450,107 @@ static TEE_Result get_sm2_kep_params(const TEE_Attribute *params,
 
 	for (n = 0; n < param_count; n++) {
 		const TEE_Attribute *p = &params[n];
+		void *bbuf = NULL;
 
 		switch (p->attributeID) {
 		case TEE_ATTR_SM2_KEP_USER:
 			kep_parms->is_initiator = !p->content.value.a;
 			found |= BIT(IS_INITIATOR);
 			break;
 		case TEE_ATTR_ECC_PUBLIC_VALUE_X:
-			crypto_bignum_bin2bn(p->content.ref.buffer,
+			res = bb_memdup_user(p->content.ref.buffer,
+					     p->content.ref.length,
+					     &bbuf);
+			if (res)
+				return res;
+
+			crypto_bignum_bin2bn(bbuf,
 					     p->content.ref.length,
 					     peer_key->x);
 			found |= BIT(PEER_KEY_X);
+			bb_free(bbuf, p->content.ref.length);
 			break;
 		case TEE_ATTR_ECC_PUBLIC_VALUE_Y:
-			crypto_bignum_bin2bn(p->content.ref.buffer,
+			res = bb_memdup_user(p->content.ref.buffer,
+					     p->content.ref.length,
+					     &bbuf);
+			if (res)
+				return res;
+
+			crypto_bignum_bin2bn(bbuf,
 					     p->content.ref.length,
 					     peer_key->y);
 			found |= BIT(PEER_KEY_Y);
+			bb_free(bbuf, p->content.ref.length);
 			break;
 		case __OPTEE_SM2_KEP_ATTR_ECC_EPHEMERAL_PUBLIC_VALUE_X:
 		case TEE_ATTR_ECC_EPHEMERAL_PUBLIC_VALUE_X:
-			crypto_bignum_bin2bn(p->content.ref.buffer,
+			res = bb_memdup_user(p->content.ref.buffer,
+					     p->content.ref.length,
+					     &bbuf);
+			if (res)
+				return res;
+
+			crypto_bignum_bin2bn(bbuf,
 					     p->content.ref.length,
 					     peer_eph_key->x);
 			found |= BIT(PEER_EPH_KEY_X);
+			bb_free(bbuf, p->content.ref.length);
 			break;
 		case __OPTEE_SM2_KEP_ATTR_ECC_EPHEMERAL_PUBLIC_VALUE_Y:
 		case TEE_ATTR_ECC_EPHEMERAL_PUBLIC_VALUE_Y:
-			crypto_bignum_bin2bn(p->content.ref.buffer,
+			res = bb_memdup_user(p->content.ref.buffer,
+					     p->content.ref.length,
+					     &bbuf);
+			if (res)
+				return res;
+
+			crypto_bignum_bin2bn(bbuf,
 					     p->content.ref.length,
 					     peer_eph_key->y);
 			found |= BIT(PEER_EPH_KEY_Y);
+			bb_free(bbuf, p->content.ref.length);
 			break;
 		case TEE_ATTR_SM2_ID_INITIATOR:
-			kep_parms->initiator_id = p->content.ref.buffer;
+			res = bb_memdup_user(p->content.ref.buffer,
```

Comment:
`bbuf` will not be freed here. No a big issue but could (unlikely) lead to bounce buffer exhaustion.

To address this, we would need a bounce buffer array passed to `get_sm2_kep_params()` and freed on function return. Is it worth it?

Reply:
I removed `bb_free()` for the same reason as above